straxen/plugins/events/event_area_per_channel.py

```diff
@@ -22,15 +22,33 @@ class EventAreaPerChannel(strax.LoopPlugin):
     n_top_pmts = straxen.URLConfig(default=straxen.n_top_pmts, type=int, help="Number of top PMTs")
 
     def infer_dtype(self):
-        dtype = [(("Area per channel for main S2", "s2_area_per_channel"),
-                  np.float32, (self.config['n_tpc_pmts'],)),
-                 (("Area per channel for alternative S2", "alt_s2_area_per_channel"),
-                  np.float32, (self.config['n_tpc_pmts'],)),
-                 (("Area per channel for main S1", "s1_area_per_channel"),
-                  np.float32, (self.config['n_tpc_pmts'],)),
-                 (("Area per channel for alternative S1", "alt_s1_area_per_channel"),
-                  np.float32, (self.config['n_tpc_pmts'],)),
-                 (("Main S1 count of contributing PMTs", "s1_n_channels"),
+        # setting data type from peak dtype
+        peak_dtype_dict = {}
+        for t_ in strax.peak_dtype(n_channels=self.config['n_tpc_pmts'], digitize_top=True):
+            peak_dtype_dict[t_[0][1]]=t_
```

Comment:
please checkout how this is done in strax: https://github.com/XENONnT/straxen/blob/35ef0abbc3e90664bce2490ae4321722a58bda6b/straxen/plugins/merged_s2s/merged_s2s.py#L67

something like `strax.unpack_dtype(self.deps['peaklets'].dtype_for('peaklets'))`

Reply:
It should be equivalent  for this particular case for peak dtype. Also, if i read the code correctly, it does not really return dictionary of name -> dtype,  but a tuple for type. So one anyway needs to construct a similar structure, and I think it looks a bit cleaner this way.

Reply:
alternatively, one can probably do: 

```python
pfields_= np.zeros(0, dtype = strax.peak_dtype(n_channels=self.config['n_tpc_pmts'], digitize_top=True)).dtype.fields
```
and then call something like
```python
dtype +=[((f'Waveform for {infoline[type_]} [ PE / sample ]', f'{type_}_data'),pfields_['data'][0])] 
```
do you think it will be better?

Reply:
No the reason I'm asking is more that the dtype of peaks changes (hypothetically) if you provide a different argument for `sum_waveform_top_array`, [see line](https://github.com/XENONnT/straxen/blob/35ef0abbc3e90664bce2490ae4321722a58bda6b/straxen/plugins/peaklets/peaklets.py#L109)


I had not looked at how exactly `strax.unpack_dtype` or `Peaklets.dtype_for` formatted the dtype (that was not the intend of my comment)